src/System.IO.FileSystem/tests/DirectoryInfo/GetSetAttributes.cs

```diff
@@ -127,5 +127,39 @@ public void WindowsInvalidAttributes_ThrowArgumentException(FileAttributes attr)
             test.Create();
             Assert.Throws<ArgumentException>(() => Set(test.FullName, attr));
         }
+
+        [ConditionalFact(nameof(CanCreateSymbolicLinks))]
+        public void SymLinksAreReparsePoints()
+        {
+            var path = GetTestFilePath();
+            var linkPath = GetTestFilePath();
+
+            Directory.CreateDirectory(path);
+            Assert.True(MountHelper.CreateSymbolicLink(linkPath, path, isDirectory: true));
+
+            Assert.NotEqual(FileAttributes.ReparsePoint, FileAttributes.ReparsePoint & Get(path));
+            Assert.Equal(FileAttributes.ReparsePoint, FileAttributes.ReparsePoint & Get(linkPath));
+        }
+
+        [ConditionalFact(nameof(CanCreateSymbolicLinks))]
+        public void SymLinksReflectSymLinkAttributes()
```

Comment:
It took me a few reads to understand this is saying that symlinks attributes are their own; but I originally read it as "symlinks reflect their target's attributes", which was the opposite intent.

Possibly consider renaming, but it's not a big deal.